apis/spaces/v1alpha1/incontrolplaneoverride_types.go

```diff
@@ -0,0 +1,293 @@
+)
+
+var (
+	regexFieldNotDeclared = regexp.MustCompile(`failed to create typed patch object.+field not declared in schema`)
+)
+
+// +kubebuilder:object:root=true
+// +kubebuilder:storageversion
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:scope=Namespaced,categories=spaces
+
+// InControlPlaneOverride represents resource configuration overrides in
+// a ControlPlane. The specified override can be applied on single objects
+// as well as claim/XR object hierarchies.
+type InControlPlaneOverride struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   InControlPlaneOverrideSpec   `json:"spec"`
+	Status InControlPlaneOverrideStatus `json:"status,omitempty"`
+}
+
+// +kubebuilder:object:root=true
+
+// InControlPlaneOverrideList is a list of InControlPlaneOverride objects.
+type InControlPlaneOverrideList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []InControlPlaneOverride `json:"items"`
+}
+
+// PropagationMode denotes the traversal direction on an object's hierarchy.
+type PropagationMode string
+
+const (
+	// PropagateAscend denotes traversal on a target object hierarchy following
+	// the metadata.ownerReferences.
+	PropagateAscend PropagationMode = "Ascending"
+	// PropagateDescend denotes traversal on a target object hierarchy
+	// following the spec.resourceRef & spec.resourceRefs reference fields.
+	PropagateDescend PropagationMode = "Descending"
+	// PropagateNone denotes that no traversal will be done and only the target
+	// object will be visited.
+	PropagateNone PropagationMode = "None"
+)
+
+// DeletionPolicy controls what happens when an InControlPlaneOverride
+// object is deleted. We either attempt to roll back the changes on the
+// target object hierarchy, or we keep them.
+type DeletionPolicy string
+
+const (
+	// DeletionRollBack attempts to roll back the changes on the target object
+	// hierarchy when the InControlPlaneOverride object is being deleted.
+	DeletionRollBack DeletionPolicy = "RollBack"
+	// DeletionKeep keeps the changes on the target object
+	// hierarchy when the InControlPlaneOverride object is being deleted.
+	DeletionKeep DeletionPolicy = "Keep"
+)
+
+// Metadata represents the Kube object metadata.
+type Metadata struct {
+	// Annotations represents the Kube object annotations.
+	// Only the following annotations are allowed to be patched:
+	// - crossplane.io/paused
+	// - spaces.upbound.io/force-reconcile-at
+	// +kubebuilder:validation:XValidation:rule="self.all(k, k == 'crossplane.io/paused' || k == 'spaces.upbound.io/force-reconcile-at')",message="Only the crossplane.io/paused and spaces.upbound.io/force-reconcile-at annotations are allowed"
+	// +kubebuilder:validation:MinProperties=1
+	// +kubebuilder:validation:MaxProperties=2
+	Annotations map[string]string `json:"annotations,omitempty"`
+}
+
+// Spec represents the spec of a Kube object.
+type Spec struct {
```

Comment:
```suggestion
type PatchSpec struct {
```

Otherwise `spacesv1alpha1.Spec` is too generic.

Reply:
Resolved. Because we are currently not targeting `spec.managementPolicies` overrides, removed `spec` from the set of allowed patches.